src/Distech.CloudRelay.API/Startup.cs

```diff
@@ -1,157 +0,0 @@
-            services.AddCors((options =>
-            {
-                options.AddDefaultPolicy(builder =>
-                {
-                    builder.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader().WithExposedHeaders(HeaderNames.ContentDisposition);
-                });
-            }));
-
-            services.AddMvc(opts =>
-            {
-                // 'AzureAD' is currently the only supported authentication provider name and anything else will allow anonymous access
-                if (IdentityProviders.AzureActiveDirectory.Equals(Configuration.GetSection(ApiEnvironment.AuthenticationProvider).Value, StringComparison.OrdinalIgnoreCase))
-                {
-                    AuthorizationPolicy policy;
-
-                    string[] roles = Configuration.GetSection(ApiEnvironment.AzureADRoles).GetChildren().Select(role => role.Value).ToArray();
-                    string[] scopes = Configuration.GetSection(ApiEnvironment.AzureADScopes).GetChildren().Select(role => role.Value).ToArray();
-
-                    if (roles.Count() > 0)
-                    {
-                        // authentication + authorization based on application permissions aka roles
-                        policy = new AuthorizationPolicyBuilder().RequireRole(roles).Build();
-                    }
-                    else if (scopes.Count() > 0)
-                    {
-                        // authentication + authorization based on delegated permissions aka scopes
-                        policy = new AuthorizationPolicyBuilder().RequireAssertion((ctx =>
-                        {
-                            var scopeClaim = ctx.User.FindFirst(c => c.Type == "http://schemas.microsoft.com/identity/claims/scope")?.Value.Split(' ');
-                            return scopeClaim != null && scopes.All(s => scopeClaim.Contains(s));
-                        }))
-                        .Build();
-                    }
-                    else
-                    {
-                        // authentication only
-                        policy = new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build();
-                    }
-
-                    opts.Filters.Add(new AuthorizeFilter(policy));
-                }
-            })
-            .SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
-
-            //allow accessing the HttpContext inside a service class
-            services.AddHttpContextAccessor();
-        }
-
-        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
-        {
-            if (env.IsDevelopment())
-            {
-                app.UseDeveloperExceptionPage();
-            }
-            else
-            {
-                app.UseHsts();
-            }
-
-            app.UseCors();
-            app.UseHttpsRedirection();
-            ConfigureAuthentication(app);
-
-            app.UseApplicationErrorHandler();
-            app.UseMultiPartRequestBuffering();
```

Comment:
Just noticed I missed that one in the migration. Did not notice any differences during my testing.
Wondering whether it could be  due to `BlobStreamDecorator` now using `async` methods because of [that breaking change in aspnetcore 3.0](https://github.com/aspnet/Announcements/issues/342).

Tried to track down the "why" and found it was added during implementation of C2D message through file request. Yet I am not sure about the intended use here.
My best/sole guess was related to telemetry purpose, until I noticed it was implemented way after we actually played with telemetry.

@usdruscia are you memories any better?

Reply:
@ggirard07 Yes, back in the .NET core 2.1 days, the request buffering was the way to go to re-read the request body in order to upload its content to blob storage (because the request body stream was already consumed by ASP.NET). There was also something called request `Rewind` but the behavior was a bit different.

Reply:
Just to confirm the behavior that storage SDK is unable to "rewind" stream to upload blob, but still creates an empty one instead, actually "silencing" the underlying issue.